### PR TITLE
Update and fix for esp8266 how to sample guide

### PR DESCRIPTION
**Pull request by momuno**
This updates the formatting of the how-to-guide to align with the other how-to-guides.
It also attempts a fix when attempting to run the `generate_arduino_zip_library.sh` on WSL.

Ignore the many, many, many, many (84!?) commits count.  It's easier to evaluate a markdown file by commiting the change and viewing it on GitHub rather than trying to use some other markdown viewing tool.

### CODE REVIEW

**Review by ericwol-msft — APPROVED** [review]

**Comment by ahsonkhan** [comment]
> It's easier to evaluate a markdown file by commiting the change and viewing it on GitHub rather than trying to use some other markdown viewing tool.

Have you tried VS Code? In my experience, it works pretty well.

File: sdk/samples/iot/docs/how_to_iot_hub_esp8266_nodemcu.md [hunk 1]
```diff
@@ -4,186 +4,201 @@
...
+        Writing at 0x00020000... (50 %)
+        Writing at 0x00024000... (55 %)
+        Writing at 0x00028000... (61 %)
+        Writing at 0x0002c000... (66 %)
+        Writing at 0x00030000... (72 %)
+        Writing at 0x00034000... (77 %)
+        Writing at 0x00038000... (83 %)
+        Writing at 0x0003c000... (88 %)
+        Writing at 0x00040000... (94 %)
+        Writing at 0x00044000... (100 %)
```
Comment (ahsonkhan) [inline sdk/samples/iot/docs/how_to_iot_hub_esp8266_nodemcu.md:164]: nit: Is all this verbatim expected output necessary to show? Can we trim it to highlight the relevant parts using `...`?


```suggestion
        Writing at 0x00000000... (5 %)
        Writing at 0x00004000... (11 %)
        ...
        Writing at 0x00040000... (94 %)
        Writing at 0x00044000... (100 %)
```

**Review by ahsonkhan — APPROVED** [review]

File: sdk/samples/iot/docs/how_to_iot_hub_esp8266_nodemcu.md (hunk 1 above)
Comment (momuno) [reply sdk/samples/iot/docs/how_to_iot_hub_esp8266_nodemcu.md:164]: It shows up only when selecting an expandable/collapsible tab, so I think it's fine in this case.

**Comment by ahsonkhan** [comment]
> Don't know about that yet. Is it an extension?

Nope, it's built-in.

This gif shows you how to use it and what it looks like:
https://code.visualstudio.com/docs/languages/markdown#_dynamic-previews-and-preview-locking

There _are_ extensions out there for both VS and VS Code that you can try if you like, but they aren't necessary.

File: sdk/samples/iot/docs/how_to_iot_hub_esp8266_nodemcu.md (hunk 1 above)
Comment (ahsonkhan) [reply sdk/samples/iot/docs/how_to_iot_hub_esp8266_nodemcu.md:164]: Good point. Another reason to make sure the `details` tag works well in doxygen and eventually the docs.microsoft.com website, once we get there :)

**Comment by momuno** [comment]
> > Don't know about that yet. Is it an extension?
> 
> Nope, it's built-in.
> 
> This gif shows you how to use it and what it looks like:
> https://code.visualstudio.com/docs/languages/markdown#_dynamic-previews-and-preview-locking
> 
> There _are_ extensions out there for both VS and VS Code that you can try if you like, but they aren't necessary.

CTRL + SHIFT + V.  WOW. the little things. :D

**Comment by ahsonkhan** [comment]
> CTRL + SHIFT + V. WOW. the little things. :D

I prefer the side-by-side view. Really makes it easy to edit and see how your changes render. Improves the inner-loop (also no more need for 84 commits :p)

**Comment by momuno** [comment]
> > CTRL + SHIFT + V. WOW. the little things. :D
> 
> I prefer the side-by-side view. Really makes it easy to edit and see how your changes render. Improves the inner-loop (also no more need for 84 commits :p)

CTRL+ K V
Works like a charm.  I think this is the best thing I've learned all week. Thank you for changing my life forever LOL!

**Pull request merged**